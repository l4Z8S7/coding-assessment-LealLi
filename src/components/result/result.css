.result {
  width: 80%;
  min-width: 300px;
  margin: auto;
  padding: 20px 30px;
  border: 5px inset #42a9d1;
  font-size: 16px;
}

.result .score {
  font-size: 24px;
  border: 2px solid #4883de;
  padding: 10px;
}

.result .score span {
  color: #f31f39;
  font-style: italic;
  font-size: 32px;
}