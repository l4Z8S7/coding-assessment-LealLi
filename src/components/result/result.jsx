import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './result.css';

export const Result = props => {
  return (
    <div className="result">
      <p>Congragulations, You have finished the assessment!</p>
      <p className="score">Your score is -- <span>{props.score}</span></p>
      <p>We will contact you when we have futher information.</p>
      <p>Thank you!</p>
    </div>
  );
}

function mapStateToProps({ questions, answers }) {
  const score = answers.map((answer, index) => answer ? questions[index].options[answer] : 0).reduce((a, b) => a + b, 0);
  return { score };
}

Result.propTypes = {
  score: PropTypes.number.isRequired
}

export default connect(mapStateToProps)(Result);