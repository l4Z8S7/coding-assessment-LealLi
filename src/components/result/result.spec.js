import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { Result } from './result';

Enzyme.configure({ adapter: new Adapter });

describe('Result', () => {
  it('should render component matching the snapshot', () => {
    expect(toJson(shallow(<Result score={50} />))).toMatchSnapshot();
  });
});