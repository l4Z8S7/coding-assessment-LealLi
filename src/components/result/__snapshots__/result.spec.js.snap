// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render component matching the snapshot 1`] = `
<div
  className="result"
>
  <p>
    Congragulations, You have finished the assessment!
  </p>
  <p
    className="score"
  >
    Your score is -- 
    <span>
      50
    </span>
  </p>
  <p>
    We will contact you when we have futher information.
  </p>
  <p>
    Thank you!
  </p>
</div>
`;
